### PR TITLE
docs: update prover and light client documentation

### PR DESCRIPTION
**Pull request by nflaig**
**Motivation**

Keep docs up-to-date and concise

**Description**

Updates docs items related to prover and light client (see commits for details)

### CODE REVIEW

File: packages/light-client/README.md [hunk 1]
```diff
@@ -18,13 +18,13 @@ The evolution of light clients is emblematic of the broader trajectory of Ethere
...
 
-Access to an beacon node that supports the light client specification is necessary. The client must support the following routes from the [consensus API spec](https://github.com/ethereum/consensus-specs/tree/dev):
+Access to an beacon node that supports the light client specification is necessary. The client must support the following routes from the [consensus API spec](https://github.com/ethereum/beacon-APIs/tree/v2.5.0/apis/beacon/light_client):
 
-- `/eth/v1/beacon/light_client/updates`
-- `/eth/v1/beacon/light_client/optimistic_update`
-- `/eth/v1/beacon/light_client/finality_update`
-- `/eth/v1/beacon/light_client/bootstrap/{block_root}`
-- `/eth/v0/beacon/light_client/committee_root`
+- `GET /eth/v1/beacon/light_client/updates`
```
Comment (matthewkeil) [inline packages/light-client/README.md]: How do you feel about making these links to the api docs?

```suggestion
- [`GET /eth/v1/beacon/light_client/updates`](https://ethereum.github.io/beacon-APIs/#/Beacon/getLightClientUpdatesByRange)
```

**Review by matthewkeil — DISMISSED** [review]
Approving but think making those routes links might be a nice touch

**Review by philknows — DISMISSED** [review]

File: packages/light-client/README.md (hunk 1 above)
Comment (nflaig) [reply packages/light-client/README.md]: great idea, we should also drop the v0, as far as I can see it's not required

**Review by philknows — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.21.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.21.0) :tada:

**Pull request merged**